frontend/src/pages/DietPlanTemplatePage.tsx

```diff
@@ -1,98 +1,118 @@
-            },
-            {
-                value: `fatsItems`,
-                btnPrompt: `הוסף שומנים`,
-                state: fatsMenueState || [],
-                sheetForm: `fats`,
-                deleteFunc: deleteMenuItem
-            },
-        ]
-    }
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string>();
 
-    useEffect(() => {
-        getMenuItems(`protein`)
-            .then(res => setProteinMenuState(res))
-            .catch(err => console.log(err))
-        getMenuItems(`carbs`)
-            .then(res => setCarbsMenuState(res))
-            .catch(err => console.log(err))
-        getMenuItems(`vegetables`)
-            .then(res => setVegetableMenuState(res))
-            .catch(err => console.log(err))
-        getMenuItems(`fats`)
-            .then(res => setFatsMenueState(res))
-            .catch(err => console.log(err))
-    }, [])
+  const tabs: ITabs = {
+    tabHeaders: [
+      {
+        name: `תפריטים`,
+        value: `dietPlanPresets`,
+      },
+      {
+        name: `חלבונים`,
+        value: `proteinItems`,
+      },
+      {
+        name: `פחמימות`,
+        value: `carbItems`,
+      },
+      {
+        name: `ירקות`,
+        value: `vegetableItems`,
+      },
+      {
+        name: `שומנים`,
+        value: `fatsItems`,
+      },
+    ],
+    tabContent: [
+      {
+        value: `dietPlanPresets`,
+        btnPrompt: `הוסף תפריט`,
+        state: dietPlanPresetsState || [],
+        sheetForm: `dietPlans`,
+        deleteFunc: deleteDietPlanPreset,
+      },
+      {
+        value: `proteinItems`,
+        btnPrompt: `הוסף חלבון`,
+        state: proteinMenuState || [],
+        sheetForm: `protein`,
+        deleteFunc: deleteMenuItem,
+      },
+      {
+        value: `carbItems`,
+        btnPrompt: `הוסף פחמימה`,
+        state: carbsMenuState || [],
+        sheetForm: `carbs`,
+        deleteFunc: deleteMenuItem,
+      },
+      {
+        value: `vegetableItems`,
+        btnPrompt: `הוסף ירקות`,
+        state: VegetableMenuState || [],
+        sheetForm: `vegetables`,
+        deleteFunc: deleteMenuItem,
+      },
+      {
+        value: `fatsItems`,
+        btnPrompt: `הוסף שומנים`,
+        state: fatsMenueState || [],
+        sheetForm: `fats`,
+        deleteFunc: deleteMenuItem,
+      },
+    ],
+  };
 
+  useEffect(() => {
+    setIsLoading(true);
+    getAllDietPlanPresets()
+      .then((res) => setDietPlanPresetsState(res))
+      .catch((err) => setError(err));
+    getMenuItems(`protein`)
+      .then((res) => setProteinMenuState(res))
+      .catch((err) => setError(err));
+    getMenuItems(`carbs`)
+      .then((res) => setCarbsMenuState(res))
+      .catch((err) => setError(err));
+    getMenuItems(`vegetables`)
+      .then((res) => setVegetableMenuState(res))
+      .catch((err) => setError(err));
+    getMenuItems(`fats`)
+      .then((res) => setFatsMenueState(res))
+      .catch((err) => setError(err));
```

Comment:
Don't you have a function that gets you all the items at once?

Reply:
Yes, but then I would have to sift them before dispersing them into the proper states so this felt easier and better to me